packages/debug-cortex/scripts/download-vscode-cortex-debug.js

```diff
@@ -0,0 +1,89 @@
+/********************************************************************************
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+const fs = require('fs');
+const https = require('https');
+const http = require('http');
+const path = require('path');
+const unzip2 = require('unzip2');
+const mkdirp = require('mkdirp');
+const packageJson = require('../package.json');
+const downloadUrl = packageJson['debugAdapter']['downloadUrl'];
+const downloadPath = path.join(__dirname, '../node_modules/download');
+const archivePath = path.join(downloadPath, path.basename(downloadUrl));
+const targetPath = packageJson['debugAdapter']['dir'];
+
+function downloadDap() {
```

Comment:
I guess it is time to move download script to some common package.

Reply:
I was thinking that, all 3 adapters have a lot of common download code. Outside of the scope of this PR IMO

Reply:
I assume that the download code is all temporary and that the Theia wrapping code will hopefully be pushed back into their respective projects?

Reply:
> I assume that the download code is all temporary and that the Theia wrapping code will hopefully be pushed back into their respective projects?

Yes, that would be ideal. IIRC the project we use was not as of yet willing to make a release available that we could consume.